web/static/js/reducers/questionnaire.js

```diff
@@ -222,26 +238,47 @@ const changeStepStore = (state, action) => {
 const addStep = (state, action) => {
   return [
     ...state,
-    newStep()
+    newMultipleChoiceStep()
   ]
 }
 
-const newStep = () => ({
-  id: uuid.v4(),
-  type: 'multiple-choice',
-  title: '',
-  store: '',
-  prompt: {
-    'en': {
-      sms: '',
-      ivr: {
-        text: '',
-        audioSource: 'tts'
+const newLanguageSelectionStep = (first: string, second: string): LanguageSelectionStep => {
+  return {
+    id: uuid.v4(),
+    type: 'language-selection',
+    title: 'Language selection',
+    store: '',
+    prompt: {
+      'en': {
+        sms: '',
+        ivr: {
+          text: '',
+          audioSource: 'tts'
+        }
       }
-    }
-  },
-  choices: []
-})
+    },
+    languageChoices: [null, first, second]
```

Comment:
I kept this from the original implementation, but I still don't understand why we create this with a `null` first element. I guess it is because we want to ensure that the default language is first there and there's some code to set it up further on. My suggestion here would be instead of doing that pass the default language as a param to this factory method. That way we can specify to Flow that `languageChoices` is an array of strings instead of an array of nullable strings, which will make a lot of subsequent code a lot less awkward.